src/codemirror/Cm6_ViewPlugin.ts

```diff
@@ -52,6 +52,34 @@ export function createCm6Plugin(plugin: ShikiPlugin): ViewPlugin<any> {
 							return;
 						}
 
+						if (props.has('inline-code')) {
+						    const content = Cm6_Util.getContent(view.state, node.from, node.to);
+						    if (content.startsWith('{')) {
+						        let match = content.match(/^\{([^\s]+)\} (.*)/i);  // format: `{lang} code`
+						        if (match) {
+						            let ranges = view.state.selection.ranges;
+						            for (const r of ranges) { // check if selection and this node overlaps
+						                if (r.from < node.to && r.to > node.from) {
+						                    this.removeDecoration(node.from, node.to);
+						                    return;
+						                }
+						            }
+						            let hideTo = node.from + match[1].length + 3;  // hide `{lang} `
+						            this.renderWidget(hideTo, node.to, match[1], match[2])
+						                .then(decorations => {
+						                    this.removeDecoration(node.from, node.to);
+						                    decorations.unshift(Decoration.replace({}).range(node.from, hideTo));
+						                    this.addDecoration(node.from, node.to, decorations);
+						                })
+						                .catch(console.error);
+						        }
+						    } else {
+						        this.removeDecoration(node.from, node.to);
+						    }
+						    return;
+						}
+						if (!docChanged) return;
```

Comment:
Why add this here and not at the start?

Reply:
Two update types are handled: `docChanged` and `selectionSet`
1. `selectionSet` affects the rendering of inline code in live preview mode. Inline code will not be highlighted when it overlaps with editor selection / cursor position.
2. It seems to me that `selectionSet` has no impact on codeblock.